File: 단계별로풀어보기/16.스택,큐,덱/01.go
// 시간 단축을 위한 풀이
// 통과

package main

import (
	"bufio"
	"fmt"
	"os"
)

var reader = bufio.NewReader(os.Stdin)
var writer = bufio.NewWriter(os.Stdout)

func main() {
	defer writer.Flush()

	stack := []int{}
	var numCommand int
	fmt.Fscan(reader, &numCommand)

	for i := 0; i < numCommand; i++ {
		var command int
		fmt.Fscan(reader, &command)

		switch command {
		case 1:
			var val int
			fmt.Fscan(reader, &val)
			stack = append(stack, val)
		case 2:
			len_stack := len(stack)
			if len_stack > 0 {
				fmt.Fprintln(writer, stack[len_stack-1])
				stack = stack[:len_stack-1]
			} else {
				fmt.Fprintln(writer, -1)
			}
		case 3:
			fmt.Fprintln(writer, len(stack))
		case 4:
			if len(stack) > 0 {
				fmt.Fprintln(writer, 0)
			} else {
				fmt.Fprintln(writer, 1)
			}
		case 5:
			len_stack := len(stack)
			if len_stack > 0 {
				fmt.Fprintln(writer, stack[len_stack-1])
			} else {
				fmt.Fprintln(writer, -1)
			}
		}
	}
}

File: 단계별로풀어보기/16.스택,큐,덱/01.py
import sys
from collections import deque

input = sys.stdin.readline
print = sys.stdout.write

stack = deque()
num_command = int(input())
res = list()

for _ in range(num_command):
    command = input().strip()
    if command[0] == "1":
        n = command[2:]
        stack.append(n)  
    elif command == "2":
        res.append(stack.pop() if stack else -1)
    elif command == "3":
        res.append(len(stack))
    elif command == "4":
        res.append(0 if stack else 1)
    elif command == "5":
        res.append(stack[-1] if stack else -1)

print("\n".join(map(str, res)))
